packages/cozy-realtime/src/CozyRealtime.js

```diff
@@ -1,360 +1,549 @@
 import MicroEE from 'microee'
-import Socket from './Socket'
-import minilog_ from 'minilog'
 
-const inBrowser = typeof window !== 'undefined'
-const minilog = (inBrowser && window.minilog) || minilog_
-const logger = minilog('cozy-realtime')
-minilog.suggest.deny('cozy-realtime', 'info')
-
-const INDEX_KEY_SEPARATOR = '//'
+import logger from './logger'
+import SubscriptionList from './SubscriptionList'
+import RetryManager from './RetryManager'
+import {
+  getUrl,
+  getToken,
+  doctype,
+  isInBrowser,
+  getCozyClientFromOptions,
+  isOnline
+} from './utils'
 
 /**
- * Generate a key for an event
- *
- * @return {String}  Event key
+ * Manage the realtime interactions with a cozy stack
  */
-export const generateKey = (eventName, type, id) =>
-  `${eventName}${INDEX_KEY_SEPARATOR}${type}${INDEX_KEY_SEPARATOR}${id}`
-
-const getHandlerAndId = (handlerOrId, handlerOrUndefined) => {
-  if (
-    typeof handlerOrId !== 'function' &&
-    typeof handlerOrUndefined !== 'function'
-  ) {
-    throw new Error('You should call this function with an handler')
+class CozyRealtime {
+  /**
+   * @constructor
+   * @param {object} options
+   * @param {CozyClient}  options.client
+   */
+  constructor(options) {
+    this.client = getCozyClientFromOptions(options)
+    this.subscriptions = new SubscriptionList()
+    this.retryManager = new RetryManager()
+    this.retryManager.on('error', err => this.emit('error', err))
+    this.bindEventHandlers()
   }
 
-  let id, handler
-  if (!handlerOrUndefined) {
-    id = undefined
-    handler = handlerOrId
-  } else {
-    id = handlerOrId
-    handler = handlerOrUndefined
+  /**
+   * Start all handlers and try to connect if online
```

Comment:
What happens if we are not online ? Will the realtime try to connect automtically ? I think this should be expressed in the class docstring.

Reply:
Nit: we usually use 3rd person in docstrings : s/Start/Starts

Reply:
:+1: for the third person, fixed.

When not online, it will connect once it receive thee online event from the browser. Let me add that in the comment